### ISSUE TITLE
Mismatch between CLIP fast and non-fast tokenizers

### ISSUE THREAD

**Issue by xenova** [huggingface/transformers#22166]
### System Info

- `transformers` version: 4.26.1
- Platform: Windows-10-10.0.22621-SP0
- Python version: 3.8.1
- Huggingface_hub version: 0.11.0
- PyTorch version (GPU?): 1.12.1+cu116 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@ArthurZucker @younesbelkada @sgugger

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Here is a minimal working example to show the mismatch:

```python
from transformers import AutoTokenizer

tokenizer_fast = AutoTokenizer.from_pretrained('openai/clip-vit-base-patch16', use_fast=True)
tokenizer = AutoTokenizer.from_pretrained('openai/clip-vit-base-patch16', use_fast=False)

text = "You should've done this"

print(tokenizer(text))
print(tokenizer_fast(text))
print(tokenizer(text) == tokenizer_fast(text))

# Outputs:
# {'input_ids': [49406, 592, 1535, 262, 563, 1700, 589, 49407], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1]}
# {'input_ids': [49406, 592, 1535, 1200, 1700, 589, 49407], 'attention_mask': [1, 1, 1, 1, 1, 1, 1]}
# False
```

It appears to stem from the `'ve` token.

### Expected behavior

The non-fast tokenization should match the fast tokenization (https://github.com/huggingface/tokenizers)

**Comment by younesbelkada** [comment]
cc @ArthurZucker

**Comment by connor-henderson** [comment]
Initially looked to fix this specific issue around apostrophes, but it became apparent there were other potential formatting inconsistencies. For example, running the below also shows differences in things like `'ll` and `!!` tokenization:  

```py
from transformers import AutoTokenizer

tokenizer_fast = AutoTokenizer.from_pretrained('openai/clip-vit-base-patch16', use_fast=True)
tokenizer = AutoTokenizer.from_pretrained('openai/clip-vit-base-patch16', use_fast=False)

text = "A\n'll !!to?'d''d of, can't."

print(tokenizer(text))
print(tokenizer_fast(text))
print(tokenizer(text) == tokenizer_fast(text))

# Outputs:
# {'input_ids': [49406, 320, 262, 865, 256, 256, 531, 286, 262, 323, 262, 262, 323, 539, 267, 753, 262, 339, 269, 49407], ...}
# {'input_ids': [49406, 320, 1342, 748, 531, 13610, 323, 8445, 323, 539, 267, 753, 713, 269, 49407], ...}
# False

```


I put up a first go at fixing this. It's ready for review but not merge until we pick a change to apply more broadly

**Comment by ArthurZucker** [comment]
Hey! 
I can't reproduce the issue, when I ran your code I got: 
```python 
{'input_ids': [49406, 592, 1535, 1200, 1700, 589, 49407], 'attention_mask': [1, 1, 1, 1, 1, 1, 1]}
{'input_ids': [49406, 592, 1535, 1200, 1700, 589, 49407], 'attention_mask': [1, 1, 1, 1, 1, 1, 1]}
```

**Comment by xenova** [comment]
I just updated to `transformers-4.27.2`, but it still produces the incorrect output. Are you sure you are running the non-fast tokenizer @ArthurZucker ?

**Comment by ArthurZucker** [comment]
I made sure that I was using fast yes 😉 
Though it is our goal to have the same output from fast and non-fast, I don't really know why there is this ftfy used here. But yes, this is most probably not used in the `fast` tokenization. This also means that the expected behaviour should probably be the one that normalized the text with `ftfy`

**Comment by xenova** [comment]
> @ArthurZucker I believe it is the opposite, the mismatch happens when ftfy is not installed. (@connor-henderson correct me if I misunderstood your posts).

Yes, this is correct. I don't have ftfy installed, and I get the mismatch.

**Comment by connor-henderson** [comment]
Re-reading the above it sounds like there's no issue as ftfy is expected to be used, closing my pr and apologies for any confusion

**Comment by ArthurZucker** [comment]
I think this PR is good for ppl who do not have `ftfy`! Thanks both of you for pointing this out and will be reviewing the PR!

**Comment by xenova** [comment]
Commenting to not mark as stale :)
